File: evaluation_machine/code/test1.cpp
#include<bits/stdc++.h>
using namespace std;
string git = "../test/git ";
string path = "./test_area/";
string command{};
int main() {
    command = git + "init -p " + path;
    cout<<">run:"<<command<<endl;
    system(command.c_str());

    printf("----------check----------\n");
    command = "tree -a " + path + ".mygit";
    cout<<">run:"<<command<<endl;
    system(command.c_str());
    return 0;
}

File: evaluation_machine/commands/test1.rs
use super::Command;
use crate::{excute, expect, run};

const GIT: &str = "../test/git";
const PATH: &str = "./test_area/";

pub struct Test1Command;

impl Command for Test1Command {
    fn execute(&self) -> super::CommandResult {
        let mut command = format!("{} init -p {}", GIT, PATH);
        excute(&command);
        
        command = format!("tree -a {}.mygit", PATH);
        excute(&command);
        
        command = format!("ls -A {}", PATH);
        if !expect(&command, |output| output.contains(".mygit"))? {
            println!("directory does not contain '.mygit'");
            return Err("directory error".into());
        }
        println!("directory contains '.mygit'");
        
        command = format!("ls -A {}.mygit", PATH);
        if !expect(&command, |output| {
            output.contains("HEAD") && output.contains("refs") && output.contains("objects")
        })? {
            println!("directory does not contain 'HEAD', 'refs', and 'objects'");
            return Err("directory error".into());
        }
        println!("directory contains 'HEAD', 'refs', and 'objects'");
        Ok(())
    }
}